saga/inMemorySagaLog.go

```diff
@@ -0,0 +1,72 @@
+package saga
+
+import "errors"
+import "fmt"
+import "sync"
+
+/*
+ * In Memory Implementation of a Saga Log, DOES NOT durably persist messages.
+ * This is for local testing purposes.
+ */
+type inMemorySagaLog struct {
+	sagas map[string]*SagaState
+	mutex *sync.RWMutex
```

Comment:
I understand why it's locking, but it's not sufficient. I'll wait to see Dan's suggested change, but what I was getting at is all locking should be moved to scheduleWork().